backend/src/packages/articles/article.repository.ts

```diff
@@ -0,0 +1,66 @@
+import { type IRepository } from '~/libs/interfaces/repository.interface.js';
+
+import { ArticleEntity } from './article.entity.js';
+import { type ArticleModel } from './article.model.js';
+
+class ArticleRepository implements IRepository {
+  private articleModel: typeof ArticleModel;
+
+  public constructor(articleModel: typeof ArticleModel) {
+    this.articleModel = articleModel;
+  }
+
+  public find(): Promise<unknown> {
+    return Promise.resolve();
+  }
+
+  public findAll(): Promise<unknown[]> {
+    return Promise.resolve([]);
+  }
+
+  public async findById(id: number): Promise<ArticleEntity | null> {
+    const article = await this.articleModel.query().findById(id).execute();
+
+    if (!article) {
+      return null;
+    }
+
+    return ArticleEntity.initialize(article);
+  }
```

Comment:
It may be done in find (there is changes in Authorization plugin PR) - so don't forget to move it to find after that changes merged

Reply:
Hi. Thanks for the review.
- I didn't do it in find, only because I wanted to have a simpler answer from the method. because in the future we will be returning full article data, with user data, comments, etc. So I decided that I didn't need that much in the context of what I did it for.

- I didn't quite understand the Authorization plugin PR, what it has to do with find. or I'm missing the context. I have certain things to do when the Authorization plugin is ready, but they have to do with getting the userId in the controller

Reply:
I've checked how you'd use it - so currently your purpose is to check that article actually exists (for that purpose we may create `checkIfExists` in repository and it would look like (`select 1 from articles where id = $id` in raw), because later on we would actually have `findById` with whole relations and it would confuse + find method in abstract repository was actually added to be extended into `find (id: number)`

Reply:
Hmmm..., a little bit wrong, I use findById for:
1. Yes, to check if the record exists in the database.
2. the received data is only the fields from this table I take as default, for transmission with payload

![image](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/assets/136931364/e976c43f-e5ef-4d99-91bd-664dd77f6eba)

Reply:
Okay, then still lets `find(id:number)` would be default findById,, and when we'd need to find with relations - create method for it separately 🤔

Reply:
- findById methods became find
- merged the branch with the "Authorization Plugin". And checked that everything was working as it should. ⬇️

![Screenshot 2023-08-25 173637](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/assets/136931364/aed6a587-6ed2-4f87-a4ec-19cc8705026b)

![Screenshot 2023-08-25 174352](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/assets/136931364/4034db80-7793-4e51-a670-7b11b3ed0590)

![image](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/assets/136931364/9a7ce086-dcca-47b1-9b3c-0f0a6c13e5f3)

```ts
[17:36:17.524] INFO (12220): POST on /api/v1/auth/sign-up
[17:43:04.418] INFO (12220): POST on /api/v1/articles/
[17:45:29.146] INFO (12220): PUT on /api/v1/articles/5
```